v2/services/authorize/credentialsmanager/managerV1.go

```diff
@@ -0,0 +1,57 @@
+package credentialsmanager
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/pkg/errors"
+
+	"github.com/SKF/go-utility/v2/log"
+)
+
+type SMAPIV1 interface {
+	GetSecretValue(input *secretsmanager.GetSecretValueInput) (*secretsmanager.GetSecretValueOutput, error)
+}
+
+type CredentialsMangerV1 struct {
+	sm SMAPIV1
+}
+
+func CreateCredentialsManagerV1(sm SMAPIV1) CredentialsManager {
```

Comment:
Exposing this interface doesn't make it super obvious that it's a AWS SDK `secretsmanager` client that should be used. I'd prefer a structure which more clearly indicates that it's AWS SDK v1/v2 that's the difference. Here it seems like there's just a v1/v2 version of the internal `credentialsmanager` type.

Wrapping the functionality in the `CredentialsManager` type is good. but I think it needs to be clear how it should be used from the naming/input of functions.

E.g `credentialsmanager.New().UsingSDKV1(sess *sesssion.Session)`? Or an option function? `credentialsmanager.New(credentialsmanager.UsingSDKV1(sess *session.Session))`.

Reply:
Fair point, I'll see what I can do about the clarity issue with credentialsmanager